connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -360,6 +391,30 @@ public void tick() {
         }
     }
 
+    private synchronized boolean checkAndMaybeRotateSessionKey(long now) {
+        if (internalRequestValidationEnabled()) {
+            if (isLeader()) {
+                boolean distributeNewKey = false;
+
+                if (sessionKey == null) {
+                    log.debug("Internal request signing is enabled but no session key has been distributed yet. "
+                        + "Distributing new key now.");
+                    distributeNewKey = true;
```

Comment:
Indirection with a variable and keeping lots of exit points from this method might be risky eventually. Documenting functionality with a variable name doesn't seem very necessary either. 

Should we keep a `return false` at the end and add two `return true` here and right below?

Reply:
Yeah, noticed that while refactoring to include changes discussed in the KIP. The variable was useless, replaced it with `return true` where appropriate.